api/lib/auth_functions.pl

```diff
@@ -41,6 +42,26 @@ sub read_json
     return decode_json($json);
 }
 
+# encode json data and write to file ($umask optional, default 022)
+# do: write_json ($filePath,$data,$umask)
+
+sub write_json {
+
+    my $filePath = shift || die 'No file path specified';
```

Comment:
Functions inside a library should avoid ending the whole script.
Please gracefully handle any error.